config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +114,184 @@ const (
 	// UserDefinedNamespaceIsolated enables monitoring for user-defined projects with namespace-scoped tenancy. This ensures that metrics, alerts, and monitoring data are isolated at the namespace level.
 	UserDefinedNamespaceIsolated UserDefinedMode = "NamespaceIsolated"
 )
+
+// alertmanagerConfig provides configuration options for the default Alertmanager instance
+// that runs in the `openshift-monitoring` namespace. Use this configuration to control
+// whether the default Alertmanager is deployed, how it logs, and how its pods are scheduled.
+//
+// When omitted, this means no opinion and the platform is left to choose a reasonable default, that is subject to change over time.
+type AlertmanagerConfig struct {
+	// deploymentMode determines whether the default Alertmanager instance should be deployed
+	// as part of the monitoring stack.
+	// Allowed values are Deployed and omitted.
+	// When set to Deployed, the Cluster Monitoring Operator
+	// ensures that an Alertmanager instance is created and managed in the `openshift-monitoring` namespace.
+	// Use this field if you want to explicitly opt in to running a platform-level Alertmanager.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default, that is subject to change over time.
+	// The current default value is `Deployed`.
+	//
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum="";Deployed
+	DeploymentMode AlertManagerDeployMode `json:"deploymentMode"`
+
+	// deployed must be set when deploymentMode is Deployed, and must be unset otherwise.
+	// When set to Deployed, the Cluster Monitoring Operator
+	// ensures that an Alertmanager instance is created and managed in the `openshift-monitoring` namespace.
+	// +optional
+	Deployed *AlertmanagerDeployedConfig `json:"deployed,omitempty"`
+}
+
+// alertmanagerConfig provides configuration options for the default Alertmanager instance
+// that runs in the `openshift-monitoring` namespace. Use this configuration to control
+// whether the default Alertmanager is deployed, how it logs, and how its pods are scheduled.
+//
+// When omitted, this means no opinion and the platform is left to choose a reasonable default, that is subject to change over time.
+type AlertmanagerDeployedConfig struct {
```

Comment:
This struct should have either:
* at least 1 required field
* a MinProperties marker